### PR TITLE
test/e2e: validate CMO and UWM configs

### PR DESCRIPTION
**Pull request by simonpasquier**
* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: test/e2e/telemeter_test.go [hunk 1]
```diff
@@ -21,30 +21,18 @@ import (
...
-			Labels: map[string]string{
-				framework.E2eTestLabelName: framework.E2eTestLabelValue,
-			},
-		},
-		Data: map[string]string{
-			"config.yaml": "{}",
-		},
-	}
-	f.MustCreateOrUpdateConfigMap(t, cm)
+	cmoConfigMap := f.BuildCMOConfigMap(t, "{xxx}")
```
Comment (simonpasquier) [inline test/e2e/telemeter_test.go:31]: the typo is on purpose to verify that BuildCMOConfigMap() fails if the input isn't valid.

File: test/e2e/telemeter_test.go [hunk 2]
```diff
@@ -21,30 +21,18 @@ import (
...
-			Labels: map[string]string{
-				framework.E2eTestLabelName: framework.E2eTestLabelValue,
-			},
-		},
-		Data: map[string]string{
-			"config.yaml": "{}",
-		},
-	}
-	f.MustCreateOrUpdateConfigMap(t, cm)
+	cmoConfigMap := f.BuildCMOConfigMap(t, "{xxx")
```
Comment (simonpasquier) [inline test/e2e/telemeter_test.go:31]: typo on purpose

Comment (simonpasquier) [reply test/e2e/telemeter_test.go:31]: which produced the expected result in the test logs:

```
 === RUN   TestTelemeterRemoteWrite
    telemeter_test.go:31: invalid CMO config yaml: line 1: did not find expected ',' or '}'
--- FAIL: TestTelemeterRemoteWrite (0.00s) 
```

**Review by sthaha — COMMENTED** [review]
/lgtm

Nice addition, @simonpasquier 🙇

**Comment by jan--f** [comment]
/label docs-approved
/label px-approved
/label qe-approved
Adding labels for test refactoring.
/lgtm
Thanks, very nice!

File: test/e2e/framework/helpers.go [hunk 3]
```diff
@@ -13,11 +14,61 @@ import (
...
+
+	E2eTestLabelName  = "app.kubernetes.io/created-by"
+	E2eTestLabelValue = "cmo-e2e-test"
+	E2eTestLabel      = E2eTestLabelName + ": " + E2eTestLabelValue
 )
 
+// BuildCMOConfigMap returns a ConfigMap holding the provided Cluster
+// Monitoring Operator's config.
+// If the config isn't valid YAML, the test will fail.
+func (f *Framework) BuildCMOConfigMap(t *testing.T, config string) *v1.ConfigMap {
```
Comment (danielmellado) [inline test/e2e/framework/helpers.go:26]: I may have missed it, sorry if it was the case, and I know all the test do that as well, but how about a 'dummy'test that just test the creation of a minimum ConfigMap and UserWorkloadConfigMap? Totally a nit,tho.

**Pull request merged**